src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -180,6 +181,13 @@ public Response getAssignments(@Context final HttpServletRequest request,
             // we want to populate gameboard details for the assignment DTO.
             for (AssignmentDTO assignment : assignments) {
                 assignment.setGameboard(gameboardsMap.get(assignment.getGameboardId()));
+
+                // Augment with group name if allowed
+                // TODO: Do we want staff and/or admins here?
+                UserGroupDTO group = groupManager.getGroupById(assignment.getGroupId());
+                if (GroupManager.isOwnerOrAdditionalManager(group, currentlyLoggedInUser.getId())) {
+                    assignment.setGroupName(group.getGroupName());
```

Comment:
This didn't appear to be firing correctly, the groupname wasn't returned for an owned group

Reply:
I can't reproduce this -- how did the problem present itself?

Reply:
I was misinterpreting what it should do/ it's purpose, it seems to work as intended

Reply:
I'm not sure I'm clear on why we need this. But as well as owners and additional managers, students can see the group name if the group last_updated is not null:
https://github.com/isaacphysics/isaac-api/blob/v2.16.1/src/main/java/uk/ac/cam/cl/dtg/segue/api/GroupsFacade.java#L214-L217

Reply:
a0f231c0 should fix this.